src/core/border_router/routing_manager.cpp

```diff
@@ -0,0 +1,771 @@
+                    mLocalOmrPrefix.ToString().AsCString(), otThreadErrorToString(error));
+    }
+    else
+    {
+        Get<NetworkData::Notifier>().HandleServerDataUpdated();
+        otLogInfoBr("published local OMR prefix %s in Thread network", mLocalOmrPrefix.ToString().AsCString());
+    }
+
+    return error;
+}
+
+void RoutingManager::UnpublishLocalOmrPrefix()
+{
+    VerifyOrExit(Get<Mle::MleRouter>().IsAttached());
+
+    IgnoreError(Get<NetworkData::Local>().RemoveOnMeshPrefix(mLocalOmrPrefix));
+    Get<NetworkData::Notifier>().HandleServerDataUpdated();
+
+    otLogInfoBr("unpubished local OMR prefix %s from Thread network", mLocalOmrPrefix.ToString().AsCString());
+
+exit:
+    return;
+}
+
+bool RoutingManager::ContainsPrefix(const Ip6::Prefix &aPrefix, const Ip6::Prefix *aPrefixList, uint8_t aPrefixNum)
+{
+    bool ret = false;
+
+    for (uint8_t i = 0; i < aPrefixNum; ++i)
+    {
+        if (aPrefixList[i] == aPrefix)
+        {
+            ret = true;
+            break;
+        }
+    }
+
+    return ret;
+}
+
+const Ip6::Prefix *RoutingManager::EvaluateOnLinkPrefix(void)
+{
+    const Ip6::Prefix *newOnLinkPrefix = nullptr;
+
+    // We don't evaluate on-link prefix if we are doing
+    // Router Discovery or we have already discovered some
+    // on-link prefixes.
+    VerifyOrExit(!mRouterSolicitTimer.IsRunning());
+    if (IsValidOnLinkPrefix(mDiscoveredOnLinkPrefix))
+    {
+        otLogInfoBr("EvaluateOnLinkPrefix: there is already on-link prefix %s on interface %s",
+                    mDiscoveredOnLinkPrefix.ToString().AsCString(), GetInfraIfName());
+        ExitNow();
+    }
+
+    newOnLinkPrefix = &mLocalOnLinkPrefix;
+
+exit:
+    return newOnLinkPrefix;
+}
+
+void RoutingManager::EvaluateRoutingPolicy(void)
+{
+    const Ip6::Prefix *newOnLinkPrefix = nullptr;
+    Ip6::Prefix        newOmrPrefixes[kMaxOmrPrefixNum];
+    uint8_t            newOmrPrefixNum = 0;
+    bool               startTimer;
+
+    OT_ASSERT(IsInitialized());
+    OT_ASSERT(Get<Mle::MleRouter>().IsAttached());
+
+    otLogInfoBr("evaluating routing policy");
+
+    // 0. Evaluate on-link & OMR prefixes.
+    newOnLinkPrefix = EvaluateOnLinkPrefix();
+    newOmrPrefixNum = EvaluateOmrPrefix(newOmrPrefixes, kMaxOmrPrefixNum);
+
+    // 1. Send Router Advertisement message if necessary.
+    startTimer = SendRouterAdvertisement(newOmrPrefixes, newOmrPrefixNum, newOnLinkPrefix);
+    if (newOmrPrefixNum == 0)
+    {
+        // This is the very exceptional case and happens only when we failed to publish
+        // our local OMR prefix to the Thread network. We schedule the Router Advertisement
+        // timer to re-evaluate our routing policy in the future.
+        otLogWarnBr("No OMR prefix advertised! Start Router Advertisement timer for future evaluation");
+        startTimer = true;
+    }
+
+    // 2. Schedule Router Advertisement timer with random interval.
+    if (startTimer)
```

Comment:
I think `startTimer` is alwasy true here. 
we can remove this `if else` blocks.